src/coreclr/vm/ceemain.cpp

```diff
@@ -569,6 +569,32 @@ void FatalErrorHandler(UINT errorCode, LPCWSTR pszMessage)
     EEPOLICY_HANDLE_FATAL_ERROR_WITH_MESSAGE(errorCode, pszMessage);
 }
 
+namespace
+{
+    void InstallStressLogOutOfMemoryRecorder()
+    {
+        static std::new_handler previous_new_handler = std::get_new_handler();
+        std::set_new_handler([]() {
+            STATIC_CONTRACT_NOTHROW;
+            // If we have not created StressLog ring buffer, we should not try to use it.
+            // StressLog is going to do a memory allocation.  We may enter an endless loop.
+            if (StressLog::t_pCurrentThreadLog != NULL)
+            {
+                STRESS_LOG_OOM_STACK_NO_SIZE();
+            }
+
+            if (previous_new_handler != nullptr)
+            {
+                previous_new_handler();
+            }
+            else
+            {
+                throw std::bad_alloc();
```

Comment:
Hmm, this makes me wonder - couldn't we just make this handler throw our OutOfMemoryException here instead of making all the changes in the exception handling macros?

Reply:
We can't as the new handler is technically process-global. So with this change, we would technically report more OOMs (as we'd report an OOM with the stack from ICU for example) but we wouldn't break a C++ library that is expecting to catch `std::bad_alloc`.

Reply:
Ah, makes sense, thank you for the explanation.

Reply:
> We can't as the new handler is technically process-global.

Does that mean that some other library loaded in the process can break us, or vice versa?

Reply:
I am wondering whether it would be better to drop this stresslogging. I do not think it is particularly important.

Reply:
I agree that is it not particularly important. Both from dump and live debugging, it will be clear that there was an OOM, so having it in the stresslog doesn't seem to provide any benefit.

Reply:
I'll remove it from this path then.